### DISCUSSION TITLE
Craft 3 non-destructive image editing?

### DISCUSSION THREAD

**Discussion by BryceBubbles** [craftcms/cms#8574] [Ideas]
Is there a way to make the new Craft 3 image editing non-destructive by having it set a transform on save? Ideally multiple transforms could be saved and re-selected, and use with the Imager plugin would be ace. So yeah this is probably feature request.

As I understand it, currently: 
* Save will destructively edit the original image, without warning. This is not the sort of power I want to give to any of my client editors.
* "Save as a new asset" will create a new image/asset. This is better, but also messy, as it will likely lead to a whole bunch of assets that a similar.

I was excited about this new feature, but if the above points are true and there's no other options I'm missing, then I can't really use it - bar focus point perhaps, but then I'd have to hide the other buttons via CPCSS plugin.

**Comment by brandonkelly** [comment]
Not currently possible (besides applying a transform to an image from your template), but I’ll mark this as a feature request.

**Comment by BryceBubbles** [comment]
@brandonkelly Thanks for the quick reply. 

Yeah transforms are very powerful. I was just hoping that the new image editing could be used to edit transforms on an image by image basis - killer feature I reckon.

In Craft 2 I use a Matrix (or Neo) field, with an image block that has a ratio field and a focus point field (plugin), all piped in to Imager transforms. Focus point plugins don't seem to be updated to Craft 3 though and maybe they wont be based on focus point being in Craft 3 image editing. 

The focus point data from image editing gets saved on the asset for use in templates ay? If so, for now I may just hide everything but the focus point setting in the new editor (using css), and still use a ratio field for some fixed crops.

Note, finding the image editor via an asset field isn't that easy. Double click thumbnail (already not an obvious action), then click the image in that popup (no indication until hover). It's definitely something we'd have to add to an instruction document for clients (although, we may hide it more for now).

**Comment by brandonkelly** [comment]
> The focus point data from image editing gets saved on the asset for use in templates ay

Yep.

> Note, finding the image editor via an asset field isn't that easy. Double click thumbnail (already not an obvious action), then click the image in that popup (no indication until hover).

Thanks for the feedback. You can also access the image editor from the Assets page, either by double-clicking on an image there (like Assets fields), or selecting an image and choosing “Edit image” from the action menu.

**Comment by BryceBubbles** [comment]
@brandonkelly Thanks again. Yeah I found image editing via the assets page "Edit image" first. But for me (maybe others?) double-click isn't an obvious action on the web, so it took me a while to find those actions.
